clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1649,30 +1649,27 @@ void SYCLIntegrationHeader::emitFwdDecl(raw_ostream &O, const Decl *D,
     auto *NS = dyn_cast_or_null<NamespaceDecl>(DC);
 
     if (!NS) {
-      if (!DC->isTranslationUnit()) {
-        const TagDecl *TD = isa<ClassTemplateDecl>(D)
-                                ? cast<ClassTemplateDecl>(D)->getTemplatedDecl()
-                                : dyn_cast<TagDecl>(D);
-
-        if (TD && !UnnamedLambdaSupport) {
-          // defined class constituting the kernel name is not globally
-          // accessible - contradicts the spec
-          const bool KernelNameIsMissing = TD->getName().empty();
-          if (KernelNameIsMissing) {
+      const TagDecl *TD = isa<ClassTemplateDecl>(D)
```

Comment:
I wonder if this needs to be lower-level than this, since would this work with 'int'?  Or the other builtin types?  I don't think they are tagdecls, but they are probably named decls.

Reply:
For the fundamental types and other builtin types, we don't even get here.  We enter here only for those types for which a forward declaration is necessary.

Reply:
I see.  @kbobrovs is most familiar with the Int header stuff, so I'll leave the review to him.

Reply:
Thanks @erichkeane